packages/opentelemetry-context-base/src/types.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** Default type for functions */
+// tslint:disable:no-any
+export type Func<T> = (...args: any[]) => T;
+
+/** A key/value object for anything that can be stored in the context */
+export interface ContextStore {
+  [key: string]: unknown;
+}
+
+export interface ContextManager {
+  /**
+   * Binds the trace context to the given function.
+   * This is necessary in order to create child spans correctly in functions
+   * that are called asynchronously (for example, in a network response
+   * handler).
+   * @param fn A function to which to bind the trace context.
+   */
+  wrapFunction<T>(fn: Func<T>): Func<T>;
+
+  /**
+   * Binds the trace context to the given event emitter.
+   * This is necessary in order to create child spans correctly in event
+   * handlers.
+   * @param emitter An event emitter whose handlers should have
+   *     the trace context binded to them.
+   */
+  wrapEmitter(emitter: NodeJS.EventEmitter): void;
```

Comment:
Given that we want to make this library compatible with the browser, what would you think about bringing in the type for NodeJS.EventEmitter inside this library?

That makes it so that we can build browser bundles with TS without having to take in the Node types. That's what I did in OpenCensus Web ([code link](https://github.com/census-instrumentation/opencensus-web/blob/a0d573f238d3d4ea6c4788517973af18d8378b05/packages/opencensus-web-types/src/node/types.ts#L24))

Reply:
I've added another comment on the new PR (https://github.com/open-telemetry/opentelemetry-js/pull/66) about this method, i don't think it's even needed with async hooks so i would be in favor of dropping it.